diff --git a/js/tutorial/sections.js b/js/tutorial/sections.js
index f7cd4cd..9527574 100644
--- a/js/tutorial/sections.js
+++ b/js/tutorial/sections.js
@@ -17,6 +17,21 @@ function register_VIM_TUTORIAL_SECTIONS(interpreter, messager, createSection, re
               messager.removeListener('pressed_key', forAbortId);
           }
       });
+      
+      if (API != null)
+      {
+        var numLevels = $('.section_menu_append_target').children().length;
+        var sectionElement = $('.section_menu_item.selected');
+        var levelId = sectionElement.index() + 1;
+        var sectionTitle = sectionElement.text()
+        
+        if (!sectionTitle.endsWith(" Γ£à"))
+        {
+          sectionElement.text(sectionTitle + " Γ£à");
+        }
+        
+        ScormSaveAnswer(levelId, numLevels);
+      }
   }
 
   function defaultPre() { interpreter.environment.setInsertMode(); }
diff --git a/js/tutorial/tutorial_engine.js b/js/tutorial/tutorial_engine.js
index cb4bc5e..759ff4b 100644
--- a/js/tutorial/tutorial_engine.js
+++ b/js/tutorial/tutorial_engine.js
@@ -4,6 +4,24 @@ function createTutorial(context, interpreter, messager, doc) {
   var tutorial_sections = [];
 
   function createSection(name, preFun, paragraphs, postFun) {
+      
+    if (API != null)
++   {
+        var sd = ScormProcessGetValue("cmi.suspend_data");
+        
+        if (sd != null)
+        {
+            var sectionElement = $('.section_menu_item.selected');
+            var levelId = String(sectionElement.index() + 1);
+            
+            var sd_arr = sd.split(",");
+            if (sd_arr.includes(levelId))
+            {
+                name = name + " Γ£à";
+            }
+        }
+    }
+      
     return {
       'name': name,
       'paragraphs': paragraphs,
@@ -19,6 +37,22 @@ function createTutorial(context, interpreter, messager, doc) {
   function showCommandOneByOne(commands, userInputAccepterCreator) {
     if(commands.length === 0) {
       sendMessageAsync('tutorial_next_section');
+      
+      if (API != null)
+      {
+        var numLevels = $('.section_menu_append_target').children().length;
+        var sectionElement = $('.section_menu_item.selected');
+        var levelId = sectionElement.index() + 1;
+        var sectionTitle = sectionElement.text()
+        
+        if (!sectionTitle.endsWith(" Γ£à"))
+        {
+          sectionElement.text(sectionTitle + " Γ£à");
+        }
+        
+        ScormSaveAnswer(levelId, numLevels);
+      }
+      
       return;
     }
 
diff --git a/tutorial.html b/tutorial.html
index e54613a..efe90e4 100644
--- a/tutorial.html
+++ b/tutorial.html
@@ -49,21 +49,11 @@
       }
     });
   </script>
+  <script src="js/scorm.js" type="text/javascript"></script>
 
-  <script type="text/javascript">
-    var _gaq = _gaq || [];
-    _gaq.push(['_setAccount', 'UA-1334187-4']);
-    _gaq.push(['_trackPageview']);
-
-    (function () {
-      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
-      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
-      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
-    })();
-  </script>
 </head>
 
-<body>
+<body onload="ScormProcessInitialize(); openNav();" onbeforeunload="ScormProcessTerminate();" onunload="ScormProcessTerminate();">
   <div id="main">
 
     <div class="navigation">
